ethereum/src/ERC20Gateway.sol

```diff
@@ -72,3 +73,24 @@ contract ERC20Gateway is IERC20Gateway, Context, IMessageQueueReceiver {
         return true;
     }
 }
+
+contract ERC20GatewayBridgingPayment is BridgingPayment {
+    constructor(
+        address _underlying,
+        address _admin,
+        uint256 _fee
+    ) BridgingPayment(_underlying, _admin, _fee) {}
+
+    /** @dev Call `requestBridging` function from `ERC20Gateway` contract. This function also
+     * deducting some fee in native tokens from such a call. For further info see `ERC20Gateway::requestBridging`.
+     */
+    function requestBridging(
+        address token,
+        uint256 amount,
+        bytes32 to
+    ) public payable {
+        deductFee();
```

Comment:
While it is very unlikely due to the logic (a call is made to the admin's address), a reentrancy attack is still possible

Reply:
Is it resolved by using `transfer` instead of `call`?

Reply:
Yes